Src/FluentAssertions/CallerIdentification/QuotesHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Text;
+
+namespace FluentAssertions.CallerIdentification
+{
+    public class QuotesHandler : IHandler
+    {
+        private readonly StringBuilder statement;
+        private char isQuoteEscapeSymbol = '\\';
+        private bool isQuoteContext;
+        private char? previousChar;
+
+        public QuotesHandler(StringBuilder statement) => this.statement = statement;
+
+        public HandlerResult Handle(char symbol)
+        {
+            if (symbol == '"')
+            {
+                if (isQuoteContext)
+                {
+                    if (previousChar != isQuoteEscapeSymbol)
+                    {
+                        isQuoteContext = false;
+                        isQuoteEscapeSymbol = '\\';
+                    }
+                }
+                else
+                {
+                    isQuoteContext = true;
+                    if (IsVerbatim())
+                    {
+                        isQuoteEscapeSymbol = '"';
+                    }
+                }
+            }
+
+            if (isQuoteContext)
+            {
+                statement.Append(symbol);
+            }
+
+            previousChar = symbol;
+            return isQuoteContext ? HandlerResult.Handled : HandlerResult.InProgress;
+        }
+
+        private bool IsVerbatim()
```

Comment:
`@$""` is also a valid interpolated verbatim string.

Reply:
I think that's what it is doing?

Reply:
Ahh sorry, I meant that `$@""` is also a valid interpolated verbatim string.

Reply:
ahh OK - I'll try to get this resolved today hopefully.

Reply:
Done